consensus/safety-rules/src/safety_rules.rs

```diff
@@ -34,7 +35,7 @@ use std::marker::PhantomData;
 /// set)
 pub struct SafetyRules<T> {
     persistent_storage: PersistentSafetyStorage,
-    validator_signer: ValidatorSigner,
+    validator_signer: Option<ValidatorSigner>,
```

Comment:
why this has to be Option?

Reply:
I guess @davidiw suggested this change because it then makes the signer type aligned with verifier (i.e. both are Option<>)... But logically this shall never be None?

Reply:
It isn't initialized until we get our first epoch ending li, otherwise how do we know which private key to retrieve from storage.

Reply:
then we probably should change the constructor too

Reply:
@zaxguo, it doesn't seem like we have refactored the constructor (new method)

Reply:
@davidiw @zekun000  Guys, how about the current one? Had to do some dirty hack in reconcile_key make it work with existing tests